### DISCUSSION TITLE
JIT mode seems to remove certain styles

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/tailwindcss#5199] [Help]
I've been experimenting a bit with JIT enabled in our Vue project and it I've been experiencing several issues with both custom and native classes being applied, but the styling doesn't take effect.

E.g. my Vue component has been given the custom class `bg-blue-100` via a prop and although the value is added dynamically to the class, Tailwind doesn't find and apply the styling, even though added a different custom class will work.

Another component receives the native class `w-6/12` as a prop and applies the class, but again the styling doesn't show. If I then pass a different native class like `w-2/4` both class and styling is applied 😯

If I remove `mode: JIT` from the tailwind.config.js all classes and stylings work as expected.

I've googled this for a week now, and tried all kinds of solutions, but none have worked.

**Tried solutions:**
- Exiting session and starting server again using `yarn dev`
- Ran `yarn build`and then `yarn dev`
- Suspected PurgeCSS might be the culprit, so removed all the lines one by one, to see if it would have an affect
- Added options and safelist/whitelist to PurgeCSS
- Created a stub.html with all the needed classes
- Added the classes to the css of the component, which worked with JIT, but isn't an option in the long run
- Added `TAILWIND_MODE--watch` to nuxt.config

So now I'm at my wits' end and hope one of you can help me along 🙏🏻

**tailwind.config.js**
    const colors = require("tailwindcss/colors");
    const plugin = require("tailwindcss/plugin");

    module.exports = {
      mode: 'jit',
      purge: {
        content: [
          './cms/*.vue',
          './components/**/*.{vue,js}',
          './composables/**/*.{js,ts}',
          './layouts/**/*.vue',
          './pages/**/*.vue',
          './plugins/**/*.{js,ts}',
          '~nuxt.config.{js,ts}'
        ],
      },
      darkMode: "class", // or 'media' or 'class'
      theme: {
        minHeight: {
          '0': '0',
          '1/4': '25%',
          '1/2': '50%',
          '3/4': '75%',
          'full': '100%',
          '72': '18rem',
          '44': '11rem',
          '28': '7rem'
        },
        fontFamily: {
          sans: ['Gilroy', 'Open Sans','Helvetica', 'Arial', 'sans-serif']
        },
        extend: {
          fontSize: {
            'xl': '1.5rem',
            '2xl': '1.8333rem',
            '3xl': '2.1667rem',
            '4xl': '2.6667rem'
          },
          lineHeight: {
            '7': '2rem',
            '8': '2.1667rem',
            '9': '2.3333rem',
            '10': '3rem'
          }
        },
        container: {
          center: true,
        },
    
        colors: {
          transparent: 'transparent',
          current: 'currentColor',
          white: colors.white,
          grey: {
            100: '#000000',
            80: '#646464',
            60: '#aeaeae',
            40: '#c8c8c8',
            20: '#e0e0e0',
            15: '#f3f3f3',
            10: '#fafafa'
          },
          pink: {
            100: '#FFD8EA',
            60: '#FFDFEE',
            20: '#FFF4F9'
          },
          blue: {
            100: '#002788',
            60: '#677EB8',
            20: '#CDD4E7'
          },
          cyan: {
            100: '#A5D3EB',
            60: '#C8E4F2',
            20: '#ECF6FA'
          },
          burgundy: {
            100: '#6E0850',
            60: '#A96C97',
            20: '#E2CDDC'
          },
          yellow: {
            100: '#FBBA1E',
            60: '#fdd678',
            20: '#fff1d2'
          },
          red: {
            100: '#F13A3A',
            60: '#F68888',
            20: '#FCD7D7'
          },
          brown: {
            100: '#A13716',
            60: '#C68773',
            20: '#ECD7D0'
          },
          lime: {
            100: '#9BC31E',
            60: '#C3DB78',
            20: '#eBF3D2'
          },
          green: {
            100: '#0D6D32',
            60: '#6DA784',
            20: '#CFE2D6'
          },
          gray: {
            darkest: '#1f2d3d',
            dark: '#3c4858',
            DEFAULT: '#c0ccda',
            light: '#e0e6ed',
            lightest: '#f9fafc',
          }
        },
    
        screens: {
          'sm': {'min': '320px', 'max': '767px'},
          // => @media (min-width: 640px) { ... }
    
          'md': {'min': '768px', 'max': '1023px'},
          // => @media (min-width: 768px) { ... }
    
          'lg': {'min': '1024px', 'max': '1279px'},
          // => @media (min-width: 1024px) { ... }
    
          'xl': {'min': '1280px', 'max': '1560px'},
          // => @media (min-width: 1280px) { ... }
    
          '2xl': {'min': '1561px'},
          // => @media (min-width: 1536px) { ... }
        },
      },
      variants: {
        extend: {
    
        },
      },
      plugins: [
        require('@tailwindcss/forms'),
      ],
    };

**Comment by Martleby** [comment]
When you're running with jit, do the classes appear in the generated css file?

I'm thinking this might be a problem/conflict between PurgeCSS and JIT, and possibly the way it's called. In my tailwind conf I set purge like this:

```
purge: [
        './resources/**/*.{vue,js,php}',
        './app/View/**/*.php',
        './app/Traits/**/*.php'

    ], 
```

So without the content object. Is that worth trying?

**Reply by ghost** [reply]
Hi @Martleby I tried that, before coming across a S/O thread where the content object was advised - but none of them does the trick.

The classes appear in the class attribute on the divs, but not in the stylesheet (at least not visible in the inspector tool):
<img width="715" alt="Screenshot 2021-08-16 at 10 39 17" src="https://user-images.githubusercontent.com/72138377/129535868-69ee1de0-7f3a-446b-b377-468e38fcd6ea.png">

Here the class `bg-blue-100` is added to the div, but doesn't appear in the styles in the inspector, but is would appear and show correctly if I turn of JIT 😑

**Comment by simonswiss** [comment]
Hey!

Those errors where random classes seem to not be applied usually signal some issue with the build tool and how PostCSS is handled.

Just checking it's not a typo, but in your message you wrote this:

> Added `TAILWIND_MODE--watch` to nuxt.config

It should be `TALIWIND_MODE=watch`, and that usually goes in your package.json file before the script your run to start your dev server / build process.

**Reply by ghost** [reply]
Hi  Simon, thank you for your answer. I'm a bit confused as to how to add this to the package.json, is this correct?

    "scripts": {
        "dev": "TALIWIND_MODE=watch nuxt",
        "build": "nuxt build -m",
        "build:analyze": "nuxt build -a -m",
        "start": "nuxt start",
        "generate": "nuxt generate",
        "test": "jest"
      },
